shell/platform/darwin/macos/framework/Source/FlutterBackingStoreDescriptor.h

```diff
@@ -0,0 +1,50 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <Cocoa/Cocoa.h>
+#import <Metal/Metal.h>
+
+/**
+ * The type of the backing store, either Metal or OpenGL.
+ */
+typedef NS_ENUM(NSInteger, FlutterMacOSBackingStoreType) {
+  FlutterMacOSBackingStoreTypeMetal,
+  FlutterMacOSBackingStoreTypeOpenGL
+};
+
+/**
+ * A wrapper around backing store handles. On OpenGL, refers to the frame buffer id, and on Metal,
+ * holds the reference to the MTLTexture. Will only contain one of Metal or OpenGL backing store
+ * handles.
+ */
+@interface FlutterBackingStoreDescriptor : NSObject
```

Comment:
I'm not sure what "Descriptor" means here; it's holding the actual backing stores, not descriptions of them.